### ISSUE TITLE
Bug in Layout/LineEndStringConcatenationIndentation

### ISSUE THREAD

**Issue by robotdana** [rubocop/rubocop#9916]
When the EnforcedStyle is aligned, and a concatenated string isn't assigned or a method arg, the second string line is indented instead. 

--------

## Expected behavior

```
block do
  'string' \
  'string'
end
```

## Actual behavior

```
block do
  'string' \
    'string'
end
```

## Steps to reproduce the problem

Run rubocop -A on this file
```
# frozen_string_literal: true

block do
  'string' \
  'string'
end

puts 'string' \
     'string'

'string' \
'string'
```

with this .rubocop.yml
```
AllCops:
  NewCops: enable

Layout/LineEndStringConcatenationIndentation:
  EnforcedStyle: aligned
```

and expect no changes

## RuboCop version

1.18.2

**Comment by robotdana** [comment]
sorry, read the documentation more clearly, this is a documented exception. ...why is it an exception?

**Comment by jonas054** [comment]
I'm the one who added `Layout/LineEndStringConcatenationIndentation`. Usually I look to the ruby-mode of Emacs for guidance on indentation, but with this cop I based it more on how `Layout/MultilineOperationIndentation` works. So that's the reason. It mimics the style here:

```ruby
block do
  'string' +
    'string'
end
```
